### DISCUSSION TITLE
0.4.5 - new branch permits decreased logging

### DISCUSSION THREAD

**Discussion by richb-hanover** [sqm-autorate/sqm-autorate#126] [Algorithm Design]
I just committed `testing/lua-threads` with a version number of 0.4.5. It has exactly the same algorithm as 0.4.4 with a new csv_logger() function that honors the `log_level`. https://github.com/sqm-autorate/sqm-autorate/commit/a566f02a15f6441353359c01d2f81a4243bb7f1a

I made this mod because my router was crashing because /tmp was filling up with log files. Setting `log_level` to "WARN" in the `/etc/config/sqm-autorate` file will skip all the volumnious logging. 

This new version warrants checking: I believe it's the same, but...

I also created a new script that retrieves debugging from an OpenWrt router. Run it from your laptop, and it'll ssh in, request the getstats.sh output and the three /tmp/sqm-* files for your troubleshooting amusement.

I also edited the setup.sh script. Its earlier form had an error if `curl` wasn't already installed. I also factored out the "Is this OpenWrt?" tests to abort before continuing on to the main routine.

**Comment by richb-hanover** [comment]
@chas-iot I see you're working on the `develop/main` branch. I want to point out that I edited the `setup.sh` script in the `testing/lua-threads` branch to fix a failure if curl wasn't already installed, and to move all the "installing on the wrong system" tests to the very start of the script. 

There are no intentional changes to the script functionality (except to add curl to the `opkg install...`) Might you look to include it?

**Comment by chas-iot** [comment]
@richb-hanover I'll get to this today, I hope. I did bounce off your changes a couple of times when I tried to integrate with mine, as yours are committed direct to `testing\lua-threads`. Your explanation above should mean that I can proceed quickly.

Best practice (which I should follow too ;) ) is to develop on a feature branch off `develop/main`. When lightly tested, then merge the feature branch into `develop/main` where it should(!) get integration tested along with all other upcoming changes. Finally, a release gets cut (several steps) and `develop/main` gets merged into `testing/lua-threads`.

**Comment by chas-iot** [comment]
@richb-hanover please review #134 

It should incorporate the spirit of your changes, if not the letter

**Reply by richb-hanover** [reply]
> Best practice (which I should follow too ;) ) is to develop on a feature branch off develop/main

Ahah! I will follow this practice going forward. Thanks.
